src/components/ForecastDetails.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import ForecastTable from './ForecastTable';
+import FetchError from './FetchError';
+import { getForecastData, getErrorMessage } from '../reducers';
+
+class ForecastDetails extends React.Component {
```

Comment:
1. This can be pure function: `const ForecastDetails = ({ forecastData, errorMessage }) => `
2. This: `errorMessage && !forecastData.length` can be function: `const hasError = () =>`
3. Information about error existance can be extracted in `mapStateToProps` and just used as `hasError` prop
4. This condition can be tertiary argument `?:`

Rule is:

If this does not require lifecycle methods or `state` there is no real reason to make it class.

Reply:
Ok, got it. This is something that I'm trying to get better at :) Thanks for pointing out.